<launch>

    <arg name="robot_name" default="sam"/>

    <node name="angle_wrapper" pkg="sam_basic_controllers" type="angle_wrapper" ns="dynamic_heading" output="screen" >
        <param name="feedback_topic" value="/$(arg robot_name)/ctrl/yaw_feedback"/>
        <param name="setpoint_topic" value="/$(arg robot_name)/ctrl/dynamic_heading/setpoint"/>
        <param name="angle_error_topic" value="/$(arg robot_name)/ctrl/dynamic_heading/angle_error"/>
        <param name="zero_topic" value="/$(arg robot_name)/ctrl/dynamic_heading/zero_setpoint"/>
        <param name="loop_freq" value="50" />
    </node>

    <node name="dyn_heading_pid_1" pkg="pid" type="controller" ns="dynamic_heading" output="screen" >
      <param name="Kp" value="1.0" />
      <param name="Ki" value="0.0" />
      <param name="Kd" value="0.01" />
      <param name="upper_limit" value="0.5" />
      <param name="lower_limit" value="-0.5" />
      <param name="windup_limit" value="100" />
      <param name="max_loop_frequency" value="100.0" />
      <param name="min_loop_frequency" value="100.0" />
      <!--CHANGE LAUNCH FILE TO CHOOSE SUBSCRIBER AND PUBLISHER TOPICS!! /-->
      <param name="topic_from_plant" value="/$(arg robot_name)/ctrl/dynamic_heading/angle_error" />
      <param name="topic_from_controller" value="r_setpoint" />
      <!-- NOTE to Harsha: this is very weird -->
      <remap from="setpoint" to="zero_setpoint" />
      <!--remap from="pitch_setpoint" to="/pitch_setpoint" /-->
      <!--remap from="/plant_state_pitch" to="state" />
      <remap from="control_effort" to="/control_effort" /-->
     </node>

     <node name="dyn_heading_pid_2" pkg="pid" type="controller" ns="dynamic_heading" output="screen" >
       <param name="Kp" value="-0.5" />
       <param name="Ki" value="-0.0" />
       <param name="Kd" value="-0.02" />
       <param name="upper_limit" value="0.1" />
       <param name="lower_limit" value="-0.1" />
       <param name="windup_limit" value="100" />
       <param name="max_loop_frequency" value="100.0" />
       <param name="min_loop_frequency" value="100.0" />
       <!--CHANGE LAUNCH FILE TO CHOOSE SUBSCRIBER AND PUBLISHER TOPICS!! /-->
       <param name="topic_from_plant" value="/$(arg robot_name)/ctrl/r_feedback" />
       <param name="topic_from_controller" value="/$(arg robot_name)/ctrl/rudder_setpoint" />
       <remap from="setpoint" to="r_setpoint" />
       <!--remap from="/plant_state_pitch" to="state" />
       <remap from="control_effort" to="/control_effort" /-->
      </node>

</launch>
